### PR TITLE
stats: fix selectivity estimation for primary key

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 
```sql
MySQL [test]> create table t (a char(5) primary key, b int);
MySQL [test]> explain select * from t where a > "a";
+---------------------+----------+------+------------------------------------------------------------+
| id                  | count    | task | operator info                                              |
+---------------------+----------+------+------------------------------------------------------------+
| TableReader_7       | 10000.00 | root | data:Selection_6                                           |
| └─Selection_6       | 10000.00 | cop  | gt(test.t.a, "a")                                          |
|   └─TableScan_5     | 10000.00 | cop  | table:t, range:[-inf,+inf], keep order:false, stats:pseudo |
+---------------------+----------+------+------------------------------------------------------------+
```

The filter condition `a > "a"` should filter some rows.
### What is changed and how it works?

This is caused by we miss taken the column `a` as handle, while it is just a primary key, so when estimating the conditions, we use the condition ranges as `int` ranges, while it is `char` ranges.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

PTAL @zz-jason @winoros @eurekaka

### CODE REVIEW

File: statistics/bootstrap.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (h *Handle) initStatsHistograms4Chunk(is infoschema.InfoSchema, tables stat
 				continue
 			}
 			hist := NewHistogram(id, ndv, nullCount, version, &colInfo.FieldType, 0, totColSize)
-			table.Columns[hist.ID] = &Column{Histogram: *hist, Info: colInfo, Count: nullCount}
+			table.Columns[hist.ID] = &Column{Histogram: *hist, Info: colInfo, Count: nullCount, isHandle: tbl.Meta().PKIsHandle && mysql.HasPriKeyFlag(colInfo.Flag)}
```
[the remark this answers is not in the document]
Comment (lamxTyler) [reply statistics/bootstrap.go]: Then the meaning of `isHandle` will be whether this table's primary key is handle, I think this is weird to put in `Column`.

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@winoros @eurekaka PTAL

**Comment by lamxTyler** [comment]
@winoros If the column type is not int, then we treat it as a normal index, so it will have exact stats like other indices.

**Pull request merged**